x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/SemanticMatchTestCase.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.qa.rest;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.ResponseException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.hamcrest.core.StringContains.containsString;
+
+public abstract class SemanticMatchTestCase extends ESRestTestCase {
+    public void testWithMultipleInferenceIds() throws IOException {
+        String query = """
+            from test-semantic1,test-semantic2
+            | where match(semantic_text_field, "something")
+            """;
+        ResponseException re = expectThrows(ResponseException.class, () -> runEsqlQuery(query));
+
+        assertThat(re.getMessage(), containsString("Field [semantic_text_field] has multiple inference IDs associated with it"));
+
+        assertEquals(400, re.getResponse().getStatusLine().getStatusCode());
+    }
+
+    public void testWithInferenceNotConfigured() {
+        String query = """
+            from test-semantic3
+            | where match(semantic_text_field, "something")
+            """;
+        ResponseException re = expectThrows(ResponseException.class, () -> runEsqlQuery(query));
+
+        assertThat(re.getMessage(), containsString("Inference endpoint not found"));
+        assertEquals(404, re.getResponse().getStatusLine().getStatusCode());
+    }
+
+    @Before
```

Comment:
Would `@BeforeClass` / `@AfterClass` be better in the setup / teardown methods as no data is modified?

Reply:
I am having a bit of trouble using `AfterClass`/`BeforeClass` even if I switch to `adminClient()`:

```
SemanticMatchIT > classMethod FAILED
    java.lang.NullPointerException: Cannot invoke "org.elasticsearch.client.RestClient.performRequest(org.elasticsearch.client.Request)" because the return value of "org.elasticsearch.xpack.esql.qa.rest.SemanticMatchTestCase.adminClient()" is null
        at __randomizedtesting.SeedInfo.seed([9885A2F1F0E88028]:0)
        at org.elasticsearch.xpack.esql.qa.rest.SemanticMatchTestCase.wipeData(SemanticMatchTestCase.java:105)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:909)
        at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:48)
```

I see that most integration tests use `@Before` and `@After`, so I will revert to that.